### PR TITLE
Proposal: Adding new debug command

### PR DESCRIPTION
**Pull request by emulienfou**
This Pull Request is a proposal to add a new debug command to help the developer to debug the settings of his application.

###  The debug:settings command displays all available settings:
  
    php bin/console debug:settings
  
###  Use the --domains option to display all configured domains:
  
    php bin/console debug:settings --domains
  
###  Display settings for a specific domain by specifying its name with the --domain option:
  
    php bin/console debug:settings --domain=default
  
###  Find all settings with a specific tag by specifying the tag name with the --tag option:
  
    php bin/console debug:settings --tag=foo

### CODE REVIEW

**Comment by martiis** [comment]
Looks nice. Please run `composer phpcs-fix` to pass travis.

sorry for late notice, busy times :)

**Comment by emulienfou** [comment]
Just having an issue when listing all settings, I have some duplicated values:

```
 -------------------- ------------------------------------------------------- --------- ---------- -------------- --------------- --------- 
  Name                 Description                                             Domain    Type       Data           Provider name   Tags     
 -------------------- ------------------------------------------------------- --------- ---------- -------------- --------------- --------- 
  site_name            The website name as it should appear on the front-end   default   string     HarmonyDemo    config          general  
  tag_line             In a few words, explain what this site is about.        default   string     test           config          general  
  meta_robots_index                                                            default   choice     index          config          seo      
  meta_robots_follow                                                           default   choice     follow         config          seo      
  meta_title                                                                   default   string     Harmony demo   config          seo      
  theme                                                                        default   choice     acme           config          theme    
  theme                                                                        default   choice                    config          theme    
  site_name            The website name as it should appear on the front-end   default   string     HarmonyCMS     config          general  
  tag_line             In a few words, explain what this site is about.        default   string                    config          general  
  meta_title                                                                   default   string                    config          seo      
  meta_description                                                             default   textarea                  config          seo      
  meta_keywords        Meta keywords should be separated by comma.             default   textarea                  config          seo      
  meta_robots_index                                                            default   choice     index          config          seo      
  meta_robots_follow                                                           default   choice     follow         config          seo      
 -------------------- ------------------------------------------------------- --------- ---------- -------------- --------------- --------- 
```
Due to the provider who can be `config` or `orm`, but in some cases it seems the provider name is `null` and return `config` in my console command.

**I will take some times do check about that**

File: src/Command/SettingsDebugCommand.php [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            // Display settings filtered by tags
+            if ($tag = $input->getOption('tag')) {
+                foreach (array_reverse($this->settingsManager->getEnabledSettingsByTag($domains,
+                    $tag)) as $settingModel) {
+                    $tableRows[] = $this->_renderSettingsRow($settingModel);
+                }
+            } // Display all settings
+            else {
+                /** @var SettingsProviderInterface $provider */
+                foreach (array_reverse($this->settingsManager->getProviders()) as $pName => $provider) {
```
Comment (martiis) [inline src/Command/SettingsDebugCommand.php]: Null provider names are possible due to this. These values is being set by SettingsManager. Options here:
 - set provider name manually to setting model just below few lines
 - use settings manager `getSettingsByDomain` but then settings will be merged.

**Comment by martiis** [comment]
👍

**Pull request merged**